ConsoleUI.js

```diff
@@ -167,45 +186,134 @@ const consoleUI = (function (readline) {
         `Insufficient funds. Player ${player.name} declares bankruptcy.`
       ),
     mapPropertyShortDisplay,
-    showPlayerTable: (players) => {
-      const propertyText = (prop) => {
-        return `${mapPropertyGroupToColor(prop)}${
-          prop.buildings === 5
-            ? ' 🏨'
-            : prop.buildings > 0
-            ? ' ' +
-              Array(prop.buildings)
-                .fill()
-                .map((_) => '🏠')
-                .join('')
-            : ''
-        } ${prop.name} - 💰: $${prop.price}${
-          prop.mortgaged ? ' | ' + c.red('mortgaged') : ''
-        }`;
-      };
+    showPlayerTradeTable: (players) => {
+      let cliTable = new Table();
+      cliTable.push([
+        '',
+        'cash',
+        'assets',
+        'tradeable properties',
+        'n/a properties',
+      ]);
 
+      players.forEach((p) => {
+        // ~ fancy use of tilde operator: https://wsvincent.com/javascript-tilde/ for reference
+        let row = [
+          `${p.name} ${~p.jailed ? '🔒' : ''}${p.bankrupt ? '💀' : ''}`,
+          `${c.green('$' + p.cash)}`,
+          `${c.green('$' + p.assets)}`,
+          `${
+            p.tradeableProps.length !== 0
+              ? p.tradeableProps
+                  .map((p) => propertyText(p))
+                  .reduce((prev, curr) => `${prev}\n${curr}`)
+              : ''
+          }`,
+          `${
+            p.untradeableProps.length !== 0
+              ? p.untradeableProps
+                  .map((p) => propertyText(p, true))
+                  .reduce((prev, curr) => `${prev}\n${curr}`)
+              : ''
+          }`,
+        ];
+        cliTable.push(row);
+      });
+
+      console.log(cliTable.toString());
+    },
+    showPlayerTable: (players) => {
       let cliTable = new Table();
       cliTable.push(['', 'position', 'cash', 'assets', 'properties']);
 
       players.forEach((p) => {
-        // !~ fancy use of tilde operator: https://wsvincent.com/javascript-tilde/ for reference
-        cliTable.push([
+        // ~ fancy use of tilde operator: https://wsvincent.com/javascript-tilde/ for reference
+        let row = [
           `${p.name} ${~p.jailed ? '🔒' : ''}${p.bankrupt ? '💀' : ''}`,
-          `${p.position} (${p.playerBoardSpace.name})`,
+          `${
+            p.playerBoardSpace
+              ? p.position + ' (' + p.playerBoardSpace.name + ')'
+              : ''
+          }`,
           `${c.green('$' + p.cash)}`,
           `${c.green('$' + p.assets)}`,
           `${
             p.properties.length !== 0
               ? p.properties
-                  .map(propertyText)
+                  .map((p) => propertyText(p))
```

Comment:
Why did we add this extra function ?

Reply:
Aside, I was reading some FP for JS and one of the "glue" functions was `unary = f => a => f(a)`, which would be good for use-cases the standard `.map` calls, usually: `.map(unary(propertyText))` (though for this case it wouldn't be needed - the point would to cull `.map`s extra parameters to the passed function to prevent unwanted arguments from leaking in, mais ouais)

Reply:
`propertyText` was changed to binary and was causing problems because of the implicit param.